### PR TITLE
[doc] Provisioning Admission Check Controller

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Provisioning Admission Check Controller documentation.
#### Which issue(s) this PR fixes:

Relates to #1136 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
/release-note-edit
```release-note
NONE
```

File: site/content/en/docs/admission-check-controllers/_index.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+title: "Admission Checks"
+linkTitle: "Admisssion Checks"
+weight: 10
+description: >
+  This section describes the Admission Checks that are currently patr of Kueue.
```
Comment (alculquicondor) [inline site/content/en/docs/admission-check-controllers/_index.md:6]:
```suggestion
  This section describes the Admission Checks that are included with Kueue.
```

File: site/content/en/docs/admission-check-controllers/_index.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+---
+Check the [Concepts](/docs/concepts) section for the generic [Admission Check](/docs/concepts/admission_check/) description.
```
Comment (alculquicondor) [inline site/content/en/docs/admission-check-controllers/_index.md:8]:
```suggestion
You can read the [Concepts](/docs/concepts) section to learn how [Admission Checks](/docs/concepts/admission_check/) influence admission.
```

File: site/content/en/docs/admission-check-controllers/provisioning.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+---
+title: "Provisioning Admission Check Controller"
+date: 2023-10-23
+weight: 1
+description: >
+  An admission check controller providing kueue integration with cluster autoscaler.
+---
+
+The Provisioning Admission Check Controller is an Admission Check Controller designed to provide Kueues integration with [Kubernetes cluster autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler). It's mainly focused on creating [ProvisioningRequests](https://github.com/kubernetes/autoscaler/blob/4872bddce2bcc5b4a5f6a3d569111c11b8a2baf4/cluster-autoscaler/provisioningrequest/apis/autoscaling.x-k8s.io/v1beta1/types.go#L41) for the workloads holding [Quota Reservation](docs/concepts/#quota-reservation) and keeping the [AdmissionCheckState](/docs/concepts/admission_check/#admissioncheckstate) in sync.
```
Comment (alculquicondor) [inline site/content/en/docs/admission-check-controllers/provisioning.md:9]:
```suggestion
The Provisioning Admission Check Controller is an Admission Check Controller designed to integrate Kueue with the [Kubernetes cluster-autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler). It's primary function is to create [ProvisioningRequests](https://github.com/kubernetes/autoscaler/blob/4872bddce2bcc5b4a5f6a3d569111c11b8a2baf4/cluster-autoscaler/provisioningrequest/apis/autoscaling.x-k8s.io/v1beta1/types.go#L41) for the workloads holding a [Quota Reservation](docs/concepts/#quota-reservation) and keeping the [AdmissionCheckState](/docs/concepts/admission_check/#admissioncheckstate) in sync.
```

File: site/content/en/docs/admission-check-controllers/provisioning.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+The controller is part of kueue and configured by `ProvisioningACC` feature gate, check the [Installation](/docs/installation/#change-the-feature-gates-configuration) guide on details on feature gate configuration.
```
Comment (alculquicondor) [inline site/content/en/docs/admission-check-controllers/provisioning.md:11]:
```suggestion
The controller is part of kueue. You can enable it by setting the `ProvisioningACC` feature gate. Check the [Installation](/docs/installation/#change-the-feature-gates-configuration) guide on details on feature gate configuration.
```

File: site/content/en/docs/admission-check-controllers/provisioning.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+## Parameters
+
+This controller uses a `ProvisioningRequestConfig` as parameters, which uses the following schema:
+
+```yaml
```
Comment (alculquicondor) [inline site/content/en/docs/admission-check-controllers/provisioning.md:17]: The OpenAPI schema is not human-friendly.

Let's start with the example and follow by describing the important fields.

File: site/content/en/docs/admission-check-controllers/provisioning.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        type: object
+```
+
+## Example
+
+### Setup
+
+{{< include "/examples/provisioning/provisioning-setup.yaml" "yaml" >}}
+
+### Job
```
Comment (alculquicondor) [inline site/content/en/docs/admission-check-controllers/provisioning.md:68]:
```suggestion
### Job using a ProvisioningRequest
```

File: site/content/en/docs/admission-check-controllers/provisioning.md (hunk 5 above)
Comment (trasc) [reply site/content/en/docs/admission-check-controllers/provisioning.md]: done

File: site/content/en/docs/admission-check-controllers/provisioning.md (hunk 6 above)
Comment (trasc) [reply site/content/en/docs/admission-check-controllers/provisioning.md]: done

File: site/content/en/docs/admission-check-controllers/provisioning.md (hunk 5 above)
Comment (alculquicondor) [reply site/content/en/docs/admission-check-controllers/provisioning.md]: > What will really help is to also publish some kind of generated API documentation.

I agree. Maybe we can investigate how to do that in the next release. That would help a lot.

Comment (alculquicondor) [reply site/content/en/docs/admission-check-controllers/provisioning.md]: https://github.com/kubernetes-sigs/kueue/issues/1249

File: site/content/en/docs/admission-check-controllers/provisioning.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+metadata:
+  name: prov-test-config
+spec:
+  provisioningClassName: queued-provisioning.gke.io
+  managedResources:
+  - nvidia.com/gpu
+```
+
+Where:
+- **provisioningClassName** - describes the different modes of provisioning the resources. Check autoscaling.x-k8s.io ProvisioningRequestSpec.ProvisioningClassName for details.
```
Comment (alculquicondor) [inline site/content/en/docs/admission-check-controllers/provisioning.md:29]:
```suggestion
- **provisioningClassName** - describes the different modes of provisioning the resources. Check `autoscaling.x-k8s.io` `ProvisioningRequestSpec.provisioningClassName` for details.
```

**Pull request merged**